plugins/scale/scale.cpp

```diff
@@ -985,8 +985,13 @@ class wayfire_scale : public wf::plugin_interface_t
                 };
 
                 add_transformer(view);
-                double view_scale = calculate_scale(view->get_wm_geometry(),
-                    scale_data[view].transformer->get_scale_padding());
+                auto geom = view->get_wm_geometry();
+                auto tr_geom = view->transform_region(geom,
+                    scale_data[view].transformer);
+                double view_scale =
+                    calculate_scale({std::max(geom.width, tr_geom.width),
```

Comment:
why do we need `max` here?

Reply:
This only takes into account the transform if the view is grown by it. This is mainly a workaround for open / close animations. E.g. if a new view opens while scale is running, the animations plugin will attach its own animation (and I can't tell if that happens before or after this point). If this is the zoom animation, then we do not want to take it into account here.

This is not a perfect solution, since another animation in the future might make the view bigger, which will result in it scaled too small (this is the case of the fire animation, but luckily, that has a z-order above scale).

Reply:
Animations should have a higher Z order anyway (>=HIGHLEVEL), so this is a bug in the animation plugin. My logic is the following:

- At the lowest level, we have transformers which change the view for a long period (force-fullscreen, wrot, forcing a size in simple-tile, etc).
- After that, we have transformers which change the view for a duration (scale)
- After that, we have transformers which have temporary animations
- Finally, we have blur since it needs to know the background

Reply:
I see; so animations need to set a high enough z-order. Would it make sense to change the `view_2D` and `view_3D` transformer classes to allow setting a custom z-order value at creation?

Reply:
> Would it make sense to change the `view_2D` and `view_3D` transformer classes to allow setting a custom z-order value at creation?

Sounds reasonable.

Reply:
I've added these changes (let me know if they should go to a separate PR instead).

This does clash with the title overlay display a bit (in case a dialog is opened / closed, the overlay is animated together with the dialog), but I don't think this is a big issue and could be fixed separately.